src/foundry/client/documents/token.d.mts

```diff
@@ -1042,6 +1530,108 @@ declare class TokenDocument extends BaseToken.Internal.CanvasDocument {
+   */
+  resize(dimensions: Partial<TokenDocument.Dimensions>, options?: Partial<Omit<TokenDocument.Database.UpdateOperation, "updates">>): Promise<boolean>;
+
+  /**
+   * Stop the movement of this Token document. The movement cannot be continued after being stopped.
+   * Only the User that initiated the movement can stop it.
+   * @returns True if the movement was or is stopped, otherwise false
+   */
+  stopMovement(): boolean;
+
+  /**
+   * Pause the movement of this Token document. The movement can be resumed after being paused.
+   * Only the User that initiated the movement can pause it.
+   * Returns a callback that can be used to resume the movement later.
+   * Only after all callbacks and keys have been called the movement of the Token is resumed.
+   * If the callback is called within the update operation workflow, the movement is resumed after the workflow.
+   * @overload
+   * @returns The callback to resume movement if the movement was or is paused,
+   *                                              otherwise null
+   * @example
+   * ```js
+   * // This is an Execute Script Region Behavior that makes the token invisible
+   * // On TOKEN_MOVE_IN...
+   * if ( !event.user.isSelf ) return;
+   * const resumeMovement = event.data.token.pauseMovement();
+   * event.data.token.toggleStatusEffect("invisible", {active: true});
+   * const resumed = await resumeMovement();
+   * ```
+   */
+  /**
+   * Pause the movement of this Token document. The movement can be resumed after being paused.
+   * Only the User that initiated the movement can pause it.
+   * Returns a promise that resolves to true if the movement was resumed by
+   * {@link foundry.documents.TokenDocument.resumeMovement | `TokenDocument#resumeMovement`} with the same key that was passed to this function.
+   * Only after all callbacks and keys have been called the movement of the Token is resumed.
+   * If the callback is called within the update operation workflow, the movement is resumed after the workflow.
+   * @overload
+   * @param key - The key to resume movement with {@link foundry.documents.TokenDocument.resumeMovement | `TokenDocument#resumeMovement`}
+   * @returns The continuation promise if the movement was paused, otherwise null
+   * @example
+   * ```js
+   * // This is an Execute Script Region Behavior of a pressure plate that activates a trap
+   * // On TOKEN_MOVE_IN...
+   * if ( event.user.isSelf ) {
+   *   event.data.token.pauseMovement(this.parent.uuid);
+   * }
+   * if ( game.user.isActiveGM ) {
+   *   const trapUuid; // The Region Behavior UUID of the trap
+   *   const trapBehavior = await fromUuid(trapUuid);
+   *   await trapBehavior.update({disabled: false});
+   *   event.data.token.resumeMovement(event.data.movement.id, this.parent.uuid);
+   * }
+   * ```
+   */
+  pauseMovement(key?: string): () => Promise<boolean> | Promise<boolean> | null;
```

Comment:
So this is equivalent to `() => (Promise<boolean> | Promise<boolean> | null)`. Did you mean `(() => Promise<boolean>) | Promise<boolean> | null`?